core/commands/storage/upload/upload/dc_repair_routing.go

```diff
@@ -0,0 +1,152 @@
+		ShortDescription: `
+Guard broadcasts repair request to potential nodes, after negotiation, nodes prepare the contracts for 
+such repair job, sign and send to the guard for confirmation `,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"request":  hostRepairRequestCmd,
+		"response": hostRepairResponseCmd,
+	},
+}
+
+var hostRepairRequestCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Negotiate with hosts for repair jobs.",
+		ShortDescription: `
+This command sends request to mining host to negotiate the repair works.`,
+	},
+	Arguments: append([]cmds.Argument{
+		cmds.StringArg("peer-id", true, false, "Host Peer ID to send repair requests."),
+	}, hostRepairResponseCmd.Arguments...), // append pass-through arguments
+	RunTimeout: 20 * time.Second,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		cfg, err := cmdenv.GetConfig(env)
+		if err != nil {
+			return err
+		}
+		if !cfg.Experimental.StorageClientEnabled {
+			return fmt.Errorf("storage client api not enabled")
+		}
+		if !cfg.Experimental.HostRepairEnabled {
+			return fmt.Errorf("host repair api not enabled")
+		}
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		api, err := cmdenv.GetApi(env, req)
+		if err != nil {
+			return err
+		}
+		pi, err := remote.FindPeer(req.Context, n, req.Arguments[0])
+		if err != nil {
+			return err
+		}
+		// Pass arguments through to host response endpoint
+		_, err = remote.P2PCallStrings(req.Context, n, api, pi.ID, "/storage/upload/dcrepair/response",
+			req.Arguments[1:]...)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+var hostRepairResponseCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Host responds to repair requests.",
+		ShortDescription: `
+This command enquires the repairer with the contract, if agrees with the contract after negotiation,
+returns the repairer's signed contract to the invoker.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("file-hash", true, false, "File hash for the host prepares to repair."),
+		cmds.StringArg("repair-shard-hash", true, false, "Shard hash for the host prepares to repair."),
+		cmds.StringArg("file-size", true, false, "Size of the repair file."),
+		cmds.StringArg("download-reward-amount", true, false, "Reward amount for download workload."),
+		cmds.StringArg("repair-reward-amount", true, false, "Reward amount for repair workload."),
+		cmds.StringArg("repair-pid", true, false, "Host Peer ID to send repair requests."),
+		cmds.StringArg("download_contract_id", true, false, " Contract ID associated with the download requests."),
+		cmds.StringArg("repair_contract_id", true, false, "Contract ID associated with the require requests."),
+	},
+	RunTimeout: 1 * time.Minute,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctxParams, err := uh.ExtractContextParams(req, env)
```

Comment:
as a async request, we should return directly to the router node here, no need to ask router node to wait for the request result.